docker/bin/zookeeperStart.sh

```diff
@@ -142,19 +142,41 @@ if [[ "$WRITE_CONFIGURATION" == true ]]; then
   echo $MYID > $MYID_FILE
   if [[ $MYID -eq $OFFSET && -z "$SEED_NODE" ]]; then
     ROLE=participant
+    # if there is an extra address provided for the node, get it to be used
+    EXTRACONFIG=$(myExtraAddress)
     echo Initial initialization of ordinal 0 pod, creating new config.
     ZKCONFIG=$(zkConfig $OUTSIDE_NAME)
     echo Writing bootstrap configuration with the following config:
     echo $ZKCONFIG
     echo $MYID > $MYID_FILE
     echo "server.${MYID}=${ZKCONFIG}" > $DYNCONFIG
+
+    prefix="server.${MYID}="
+    suffix=";${CLIENT_PORT}"
+    if [ -n "$EXTRACONFIG" ]; then
+      echo "Extra server addresses present"
+      ORIGINALADDRESS=${ZKCONFIG%"$suffix"}
+      echo "server.${MYID}=${ORIGINALADDRESS}|${EXTRACONFIG}" > $DYNCONFIG
+    else
```

Comment:
we haven't used this before - is this required for cases where EXTRACONFIG is empty?

Reply:
It is not. When you run zkconfig it sends out a format that is <address>:<quorum-port>:<leade-port>:<role>;<client--port> which follows the zk formatting for addresses. When you have multiple addresses the ports need to be duplicated, the role is optional but you can only have ";<client-port>" once at the end of the line. So if there is no extra address <address>:<quorum-port>:<leade-port>:<role>;<client--port> works, but adding that extra address we need to trim off the ;<client-port> in the first one so we have <address1>:<quorum-port>:<leade-port>:<role>|<address2>:<quorum-port>:<leade-port>:<role>;<client--port>.